File: 389.find-the-difference.py
#
# @lc app=leetcode id=389 lang=python3
#
# [389] Find the Difference
#

# Time:  O(n)
# Space: O(1)

# @lc code=start
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        res = 0
        for c in s:
            res ^= ord(c)
        for c in t:
            res ^= ord(c)
        return chr(res)

        lookup = [0] * 26
        for c in s:
            lookup[ord(c) - 97] += 1
        for c in t:
            lookup[ord(c) - 97] -= 1
            if lookup[ord(c) - 97] < 0:
                return c


# @lc code=end

File: 389.find-the-difference.go
/*
 * @lc app=leetcode id=389 lang=golang
 *
 * [389] Find the Difference
 */

package leetcode

// @lc code=start
func findTheDifference(s string, t string) byte {
	res := byte(0)
	for i := 0; i < len(s); i++ {
		res ^= s[i]
	}
	for i := 0; i < len(t); i++ {
		res ^= t[i]
	}
	return res
}

// @lc code=end
